pdfrw/objects/pdfstring.py

```diff
@@ -13,62 +77,83 @@ class PdfString(str):
         defaults to being a direct object.
     '''
     indirect = False
-    unescape_dict = {'\\b': '\b', '\\f': '\f', '\\n': '\n',
-                     '\\r': '\r', '\\t': '\t',
-                     '\\\r\n': '', '\\\r': '', '\\\n': '',
-                     '\\\\': '\\', '\\': '', '\\(': '(', '\\)': ')'
-                     }
-    unescape_pattern = (r'(\\\\|\\b|\\f|\\n|\\r|\\t'
-                        r'|\\\r\n|\\\r|\\\n|\\[0-9]{3}|\\)')
-    unescape_func = re.compile(unescape_pattern).split
-
-    hex_pattern = '([a-fA-F0-9][a-fA-F0-9]|[a-fA-F0-9])'
-    hex_func = re.compile(hex_pattern).split
-
-    hex_pattern2 = ('([a-fA-F0-9][a-fA-F0-9][a-fA-F0-9][a-fA-F0-9]|'
-                    '[a-fA-F0-9][a-fA-F0-9]|[a-fA-F0-9])')
-    hex_func2 = re.compile(hex_pattern2).split
-
-    hex_funcs = hex_func, hex_func2
-
-    def decode_regular(self, remap=chr):
-        assert self[0] == '(' and self[-1] == ')'
-        mylist = self.unescape_func(self[1:-1])
-        result = []
-        unescape = self.unescape_dict.get
-        for chunk in mylist:
-            chunk = unescape(chunk, chunk)
-            if chunk.startswith('\\') and len(chunk) > 1:
-                value = int(chunk[1:], 8)
-                # FIXME: TODO: Handle unicode here
-                if value > 255:
-                    value = 255
-                chunk = remap(value)
-            if chunk:
-                result.append(chunk)
-        return ''.join(result)
-
-    def decode_hex(self, remap=chr, twobytes=False):
-        data = ''.join(self.split())
-        data = self.hex_funcs[twobytes](data)
-        chars = data[1::2]
-        other = data[0::2]
-        assert (other[0] == '<' and
-                other[-1] == '>' and
-                ''.join(other) == '<>'), self
-        return ''.join([remap(int(x, 16)) for x in chars])
-
-    def decode(self, remap=chr, twobytes=False):
-        if self.startswith('('):
-            return self.decode_regular(remap)
+    escapes = {
+        b'\n': b'\\n',
+        b'\r': b'\\r',
+        b'\t': b'\\t',
+        b'\b': b'\\b',
+        b'\f': b'\\f',
+        b'(' : b'\\(',
+        b')' : b'\\)',
+        b'\\': b'\\\\'
+    }
+
+    unescapes = {v[1]: k for k, v in escapes.items()}
```

Comment:
I don't think this will work quite as you want on Python 3: indexing a single item out of a bytes string gives you an integer, so `b'\\n'[1] == 110`.

Assuming you want bytes strings out, slicing will do the trick: `b'\\n'[1:] == b'n'`.